### PR TITLE
Implemented auth0 register functionality

### PR DESCRIPTION
**Pull request by pschick**
- User can now register with his email and password
- User receives confirmation after successful registering

### CODE REVIEW

File: Sources/HipMobileUI/ViewModels/Views/RegisterScreenViewModel.cs [hunk 1]
```diff
@@ -23,125 +23,120 @@
...
+
+            if (String.IsNullOrWhiteSpace(Email) && (String.IsNullOrWhiteSpace(Password) || String.IsNullOrWhiteSpace(RepeatPassword)))
+
+                DisplayEmptyEmailAndPasswordErrorMessage();
+
+            else if (String.IsNullOrWhiteSpace(Email))
+
+                DisplayEmptyEmailErrorMessage();
+
+            else if (String.IsNullOrWhiteSpace(Password) || String.IsNullOrWhiteSpace(RepeatPassword))
```
Comment (lederr) [inline Sources/HipMobileUI/ViewModels/Views/RegisterScreenViewModel.cs]: I would separate both cases for more precise user information.

File: Sources/HiPMobile/ServiceAccessLayer/AuthenticationApiAccess/AuthApiAccess.cs [hunk 2]
```diff
@@ -51,21 +51,25 @@ public AuthApiAccess(IContentApiClient clientApiClient)
...
+                Constants.BasicClientId,
+                new KeyValuePair<string, string> ("email", username),
+                new KeyValuePair<string, string> ("password", password) 
             });
-            IContentApiClient client = new ContentApiClient();
-            var result = await client.PostRequestFormBased(ServerEndpoints.LoginUrl, content);
+
+            var result = await clientApiClient.PostRequestFormBased(ServerEndpoints.RegisterUrl, content);
+
+            string jsonPayload = await result.Content.ReadAsStringAsync();
```
[the remark this answers is not in the document]
Comment (pschick) [reply Sources/HiPMobile/ServiceAccessLayer/AuthenticationApiAccess/AuthApiAccess.cs]: I have removed the unused variable

File: Sources/HipMobileUI/ViewModels/Views/RegisterScreenViewModel.cs (hunk 1 above)
Comment (pschick) [reply Sources/HipMobileUI/ViewModels/Views/RegisterScreenViewModel.cs]: This has nothing to do with changing the new registering workflow. We could open a new issue/bug for this.

**Comment by pschick** [comment]
I have removed the unused payload, now everything should be fine.

**Comment by lederr** [comment]
Error cases work still fine, but I can't register anymore. From the logical side, this should still work, but since I can't register anymore, I have to figure out if it is only because I registered before or if the call of ReadAsStringAsync(); influences other parts like the result's properties.

**Comment by lederr** [comment]
ReadAsStringAsync(); was only for debugging purposes. I can't check the successful registration once again since I'm already registered and I can't remove my account. However, I assume it's working since
- I could register successfully once before this debugging-method has been removed
- the jsonpayload variable is unused anyway
- (hopefully) result seems to be unchanged.
If I'm wrong, then either create a bug report or undo the removement of this line.

**Pull request merged**